fastai/basic_train.py

```diff
@@ -237,31 +238,34 @@ def export(self, fname:PathOrStr='export.pkl', destroy=False):
             xtra = dict(normalize=self.data.norm.keywords) if getattr(self.data, 'norm', False) else {}
             state['data'] = self.data.valid_ds.get_state(**xtra)
             state['cls'] = self.__class__
-            try_save(state, self.path, fname)
+            try_save(state, self.path, file)
         if destroy: self.destroy()
 
-    def save(self, name:PathOrStr, return_path:bool=False, with_opt:bool=True):
-        "Save model and optimizer state (if `with_opt`) with `name` to `self.model_dir`."
-        self._test_writeable_path()
+    def save(self, file:PathLikeOrBinaryStream=None, return_path:bool=False, with_opt:bool=True):
+        """Save model and optimizer state (if `with_opt`) with `file` to `self.model_dir`.
+        If `file` is a binary stream (buffer or file), write to it."""
+        if is_pathlike(file): self._test_writeable_path()
         if rank_distrib(): return # don't save if slave proc
-        path = self.path/self.model_dir/f'{name}.pth'
+        target = self.path/self.model_dir/f'{file}.pth' if is_pathlike(file) else file
         if not hasattr(self, 'opt'): with_opt=False
         if not with_opt: state = get_model(self.model).state_dict()
         else: state = {'model': get_model(self.model).state_dict(), 'opt':self.opt.state_dict()}
-        torch.save(state, path)
-        if return_path: return path
+        torch.save(state, target)
+        if return_path: return target
 
     def dl(self, ds_type:DatasetType=DatasetType.Valid):
         "Return DataLoader for DatasetType `ds_type`."
         return self.data.dl(ds_type)
 
-    def load(self, name:PathOrStr, device:torch.device=None, strict:bool=True, with_opt:bool=None, purge:bool=True,
-            remove_module:bool=False):
-        "Load model and optimizer state (if `with_opt`) `name` from `self.model_dir` using `device`."
+    def load(self, file:PathLikeOrBinaryStream=None, device:torch.device=None, strict:bool=True,
+             with_opt:bool=None, purge:bool=True, remove_module:bool=False):
+        """Load model and optimizer state (if `with_opt`) `file` from `self.model_dir` using `device`.
+        If `file` is a binary stream (buffer or file), read from it."""
```

Comment:
as the comment above. Probably just make it brief `file can be a buffer` or something like that.